### PR TITLE
add task solution

### PR DESCRIPTION
**Pull request by Vayts**
https://Vayts.github.io/layout_calendar/

### CODE REVIEW

File: .stylelintrc.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module.exports = {
   extends: [
-    "stylelint-config-recommended-scss",
+    // "stylelint-config-recommended-scss",
```
Comment (roman-mirzoian) [inline .stylelintrc.js]: Why did you decide to comment on this rule?

File: src/styles/main.scss [hunk 2]
```diff
@@ -1,3 +1,68 @@
+$day-start: (
+  "mon": 0,
+  "tue": 101px,
```
Comment (roman-mirzoian) [inline src/styles/main.scss]: The pixel values for the start day are hardcoded. It is better to try to use flex technology and not hardcode these values.

File: src/index.html [hunk 3]
```diff
@@ -7,6 +7,38 @@
     <link rel="stylesheet" href="./styles/main.scss">
   </head>
   <body>
-    <h1>Calendar</h1>
+    <ul class="calendar calendar--start-day-mon calendar--month-length-28">
```
Comment (roman-mirzoian) [inline src/index.html]: The task says: `Create a markup for the calendar block with 31 days inside`, and you now have 28 days.

**Review by roman-mirzoian — CHANGES_REQUESTED** [review]
The task is well done and the application visually works well, but it needs some finishing touches.

File: src/index.html [hunk 4]
```diff
@@ -7,6 +7,38 @@
     <link rel="stylesheet" href="./styles/main.scss">
   </head>
   <body>
-    <h1>Calendar</h1>
+    <ul class="calendar calendar--start-day-fri calendar--month-length-31">
```
Comment (Anna-Tkachenko) [inline src/index.html:10]: It's nice to see semantic tags usage

File: src/styles/main.scss [hunk 5]
```diff
@@ -1,3 +1,71 @@
...
+
 body {
   margin: 0;
 }
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  list-style: none;
+  margin: 0;
```
Comment (Anna-Tkachenko) [inline src/styles/main.scss]: We can't set margins to bem block

File: src/styles/main.scss [hunk 6]
```diff
@@ -1,3 +1,71 @@
...
+      &:nth-child(#{$i})::before {
+        content: "#{$i}";
+        font-size: 30px;
+      }
+    }
+
+    &:hover {
+      cursor: pointer;
+      transform: translateY(-20px);
+      transition: all 0.5s;
```
Comment (Anna-Tkachenko) [inline src/styles/main.scss]: Check point 9 in checklist

**Review by Anna-Tkachenko — CHANGES_REQUESTED** [review]
Add demo link to pull request description

File: src/styles/main.scss [hunk 7]
```diff
@@ -1,3 +1,69 @@
...
+$day-border: 1px;
+$day-start: (
+  0: "mon",
+  1: "tue",
+  2: "wed",
+  3: "thu",
+  4: "fri",
+  5: "sat",
+  6: "sun",
+);
```
Comment (roman-mirzoian) [inline src/styles/main.scss]: It would be cool to keep the variables in a separate file.

File: src/styles/main.scss [hunk 8]
```diff
@@ -1,3 +1,69 @@
...
+
+* {
+  box-sizing: border-box;
+}
+
+html {
+  font-family: "Arial", sans-serif;
```
Comment (roman-mirzoian) [inline src/styles/main.scss]: image

**Review by roman-mirzoian — CHANGES_REQUESTED** [review]
Good job, but there is quite a bit of work to be done.

File: src/index.html (hunk 4 above)
Comment (roman-mirzoian) [inline src/index.html:10]: Do not leave comments from mentors unanswered.

**Review by roman-mirzoian — APPROVED** [review]
Well done, but do not leave comments from mentors unanswered.

**Pull request closed without merging**